pkg/volmount/getvolumepathnamesforvolumename.go

```diff
@@ -0,0 +1,68 @@
+package volmount
+
+import (
+	"path/filepath"
+	"syscall"
+	"unicode/utf16"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// utf16ToStringArray returns the UTF-8 encoding of the sequence of UTF-16 sequences s,
+// with a terminating NUL removed. The sequences are terminated by an additional NULL.
+func utf16ToStringArray(s []uint16) (result []string) {
+	start := 0
+
+	for i, v := range s {
+		if v != 0 {
+			continue
+		}
+
+		if start == i {
+			// Empty string, that's the end.
+			return
+		}
+
+		result = append(result, string(utf16.Decode(s[start:i])))
+		start = i + 1
+	}
+
+	// The buffer was incomplete, the expected "additional NULL" wasn't present.
+	// Just return the complete values we did decode, assume the caller intended
+	// to do this, e.g. as a best-effort call without resizing the buffer.
+	return
+}
+
+// GetVolumePathNamesForVolumeName returns a list of mount points for the volumePath
+// (in format '\\?\Volume{GUID}).
+// https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumepathnamesforvolumenamew
+func GetVolumePathNamesForVolumeName(volumePath string) ([]string, error) {
+	// Must end in a backslash
+	slashedVolume := filepath.Clean(volumePath)
+	if slashedVolume[len(slashedVolume)-1] != filepath.Separator {
+		slashedVolume = slashedVolume + string(filepath.Separator)
+	}
+
+	volumeP, err := windows.UTF16PtrFromString(slashedVolume)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to utf16-ise %s", slashedVolume)
+	}
+
+	bufferLength := uint32(256)
```

Comment:
Also, just to be consistent with how such syscall's (which return the size of the required buffer if given size is not enough) are handled in rest of the code, let us set this `bufferLength` to `0` initially and then set it to whatever value we get from the syscall.

Reply:
Sure, will do.